src/postfork.cpp

```diff
@@ -101,7 +101,7 @@ bool set_child_group(job_t *j, process_t *p, int print_errors) {
             }
         }
     } else {
-        j->pgid = getpid();
+        j->pgid = getpgrp();
```

Comment:
This is definitely the right change to make - it is possible for fish to not be the process group leader, which means that the fish and the OS get different ideas about what process group a job is in. Sadness ensues. You can see this today with `echo | fish -c 'sleep 500 &; sleep 500 &; jobs -g; ps -O pgid -p (pgrep sleep)'`.